src/s1ap/handlers/erab_mod_ind.c

```diff
@@ -0,0 +1,111 @@
+/*
+* Copyright 2020-present, Infosys Ltd.
+*
+* SPDX-License-Identifier: Apache-2.0
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+
+#include "ipc_api.h"
+#include "s1ap.h"
+#include "sctp_conn.h"
+#include "s1ap_msg_codes.h"
+#include "msgType.h"
+
+extern ipc_handle ipc_S1ap_Hndl;
+
+void dump_erab_mod_info(struct s1_incoming_msg_data_t *msg)
+{
+   log_msg(LOG_INFO, "MME-UE-S1AP-ID %d\n", msg->ue_idx);
+   log_msg(LOG_INFO, "eNB-UE-S1AP-ID %d\n", msg->s1ap_enb_ue_id);
+   log_msg(LOG_INFO, "erab_mod_list_count %d\n", msg->msg_data.erab_mod_ind_Q_msg_m.erab_to_be_mod_list.count);
+   for(int i = 0; i < msg->msg_data.erab_mod_ind_Q_msg_m.erab_to_be_mod_list.count; i++)
+   {
+       log_msg(LOG_INFO,"erab_to_be_modified_item[%d].erab_id: %d\n", i, msg->msg_data.erab_mod_ind_Q_msg_m.erab_to_be_mod_list.
+		       erab_to_be_mod_item[i].e_RAB_ID);
+       log_msg(LOG_INFO,"erab_to_be_modified_item[%d].transportLayerAddress: %x\n", i, msg->msg_data.erab_mod_ind_Q_msg_m.erab_to_be_mod_list.
+                       erab_to_be_mod_item[i].transportLayerAddress);
+       log_msg(LOG_INFO,"erab_to_be_modified_item[%d].dl_gtp_teid: %d\n", i, msg->msg_data.erab_mod_ind_Q_msg_m.erab_to_be_mod_list.
+                       erab_to_be_mod_item[i].dl_gtp_teid);
+   }
+
+}
+int erab_mod_indication_handler(InitiatingMessage_t *msg)
+{
+    struct s1_incoming_msg_data_t erab_mod_ind = {0};
+    struct proto_IE erab_mod_ind_ies = {0};
+
+    int decode_status = convertErabModIndToProtoIe(msg, &erab_mod_ind_ies);
+    if (decode_status < 0)
+    {
+        log_msg(LOG_ERROR, "Failed to decode ERAB Modification Indication\n");
+
+        if (erab_mod_ind_ies.data != NULL)
+            free(erab_mod_ind_ies.data);
+
+        return E_FAIL;
+    }
+
+    for (int i = 0; i < erab_mod_ind_ies.no_of_IEs; i++)
+    {
+        switch (erab_mod_ind_ies.data[i].IE_type)
+        {
+        case S1AP_IE_MME_UE_ID:
+        {
+            log_msg(LOG_INFO,
+                    "ERAB Modification Indication S1AP_IE_MME_UE_ID.\n");
```

Comment:
print MME_ID in log. Same comment for other IEs as well.

Reply:
Done.